magento/pwa-studio
# Cypress Docker Support

## Description

This PR created a new script in the cypress tests directory that will help one with running and updating cypress test snapshots while running locally.

### run-tests
Usage: `yarn run test:ci <URL> <UPDATE_SNAPSHOTS_FLAG>` or direct invocation `./run-tests <URL> <UPDATE_SNAPSHOTS_FLAG>`
Runs your cypress tests in headless mode against the URL provided and snapshots present in the package.

If the `UPDATE_SNAPSHOTS_FLAG` is `false` (default value), snapshots won't be updated. This is the default configuration that will be used to run tests in the CI job.

If the `UPDATE_SNAPSHOTS_FLAG` is `true`, snapshots will be updated. 

Should also merge https://github.com/magento-commerce/pwa-studio-cicd/pull/87 while merging this.

## Related Issue

Closes PWA-1975

### Verification Stakeholders

@sharkySharks 

## Verification Steps
1. Check out the branch and run the `yarn test:ci <URL>` command with URL set to https://pr-3401.pwa-venia.com/
2. All tests should pass (row video test might fail which keeps failing on me every now and then, seems like a fragile test. should be updated)

## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
